torchgeo/datasets/seasonet.py

```diff
@@ -0,0 +1,495 @@
+
+import os
+import random
+from collections.abc import Callable, Collection, Iterable
+from typing import Optional
+
+import matplotlib.patches as mpatches
+import matplotlib.pyplot as plt
+import numpy as np
+import rasterio
+import torch
+from matplotlib.colors import ListedColormap
+from matplotlib.figure import Figure
+from rasterio.enums import Resampling
+from torch import Tensor
+
+from .geo import NonGeoDataset
+from .utils import download_url, extract_archive, percentile_normalization
+
+
+class SeasoNet(NonGeoDataset):
+    """SeasoNet Semantic Segmentation dataset.
+
+    The `SeasoNet <https://doi.org/10.5281/zenodo.5850306>`__ dataset consists of
+    1,759,830 multi-spectral Sentinel-2 image patches, taken from 519,547 unique
+    locations, covering the whole surface area of Germany. Annotations are
+    provided in the form of pixel-level land cover and land usage segmentation
+    masks from the German land cover model LBM-DE2018 with land cover classes
+    based on the CORINE Land Cover database (CLC) 2018. The set is split into
+    two overlapping grids, consisting of roughly 880,000 samples each, which are
+    shifted by half the patch size in both dimensions. The images in each of the
+    both grids themselves do not overlap.
+
+    Dataset format:
+
+    * images are 16-bit GeoTiffs, split into seperate files based on resolution
+    * images include 12 spectral bands with 10, 20 and 60 m per pixel resolutions
+    * masks are single-channel 8-bit GeoTiffs
+
+    Dataset classes:
+
+    0. Continuous urban fabric
+    1. Discontinuous urban fabric
+    2. Industrial or commercial units
+    3. Road and rail networks and associated land
+    4. Port areas
+    5. Airports
+    6. Mineral extraction sites
+    7. Dump sites
+    8. Construction sites
+    9. Green urban areas
+    10. Sport and leisure facilities
+    11. Non-irrigated arable land
+    12. Vineyards
+    13. Fruit trees and berry plantations
+    14. Pastures
+    15. Broad-leaved forest
+    16. Coniferous forest
+    17. Mixed forest
+    18. Natural grasslands
+    19. Moors and heathland
+    20. Transitional woodland/shrub
+    21. Beaches, dunes, sands
+    22. Bare rock
+    23. Sparsely vegetated areas
+    24. Inland marshes
+    25. Peat bogs
+    26. Salt marshes
+    27. Intertidal flats
+    28. Water courses
+    29. Water bodies
+    30. Coastal lagoons
+    31. Estuaries
+    32. Sea and ocean
+
+    If you use this dataset in your research, please cite the following paper:
+
+    * https://doi.org/10.1109/IGARSS46834.2022.9884079
+
+    .. note::
+       This dataset requires the following additional library to be installed:
+
+       * `pandas <https://pypi.org/project/pandas/>`_ to load CSV files
+
+    .. versionadded:: 0.5
+    """
+
+    metadata = [
+        {
+            "name": "spring",
+            "ext": ".zip",
+            "url": "https://zenodo.org/api/files/e2288446-9ee8-4b2e-ae76-cd80366a40e1/spring.zip",  # noqa: E501
```

Comment:
Personally I would save a base url like:
```python
url = "https://zenodo.org/api/files/e2288446-9ee8-4b2e-ae76-cd80366a40e1/"
```
and then you can append only when you need the full URL. Then you don't have so much code duplication. Not sure if you really need `ext` either, could just infer that from the filename.